### PR TITLE
Added pre-commit linting and formatting (black + flake8 + more)

### PR DESCRIPTION
**Pull request by sondrelg**
- Added pre-commit config which when installed will run several hooks before letting you commit changes to a branch. The hooks include automatic code formatting (black), linting (flake8), and other useful smaller hooks like `double-quote-string-fixer` and `end-of-file-fixer`.
- Added flake8 back into the `pythonpackage.yml` workflow.
- Added `CONTRIBUTING.rst` including instructions on how to install `pre-commit`.

Remaining changes are the result of running pre-commit on the entire project and fixing flagged errors.

To install pre-commit, simply run
- `pip install pre-commit`
- `pre-commit install`

After installation it will lint your git diff before letting you commit changes.

### CODE REVIEW

File: demoproj/settings.py [hunk 1]
```diff
@@ -109,18 +103,11 @@
...
     },
     'loggers': {
-        'django': {
-            'handlers': ['console'],  # <-- Specify handlers
-            'level': 'INFO'
-        },
-        'demoproj': {
-            'handlers': ['console'],  # <-- Specify handlers
-            'level': 'DEBUG'
+        'django': {'handlers': ['console'], 'level': 'INFO'},  # <-- Specify handlers
```
Comment (JonasKs) [inline demoproj/settings.py]: Hmmm. Can we allow multiline JSON? (Maybe only in this file) `DJANGO_GUID` above also feels less readable with this style.

Comment (sondrelg) [reply demoproj/settings.py]: I agree configurations like these are more readable when expanded. Unfortunately, I don't know that it is possible to configure black to this extent.

It is possible to exclude files by adding them to `exclude` in .black

Comment (sondrelg) [reply demoproj/settings.py]: Excluded the settings in 2f347fb5546690b3ce78230d687fa330456c7bc9

**Comment by JonasKs** [comment]
Perfect! Thank you 🍻  Closing #15

**Pull request merged**